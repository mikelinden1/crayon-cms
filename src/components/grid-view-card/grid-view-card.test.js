import GridViewCard from './grid-view-card';

const props = {
    item: { id: 1, name: 'item 1' },
    config: {
        id: 'widgets',
        moduleName: 'Manage Widgets',
        itemName: 'Widget',
        itemNamePlural: 'Widgets',
        capabilities: {
            reorderable: false,
            editable: true,
            deleteable: true,
            deleteIdProp: 'name',
            bulkImport: true
        },
        itemsPerPage: 6,
        views: {
            displayType: 'grid',
            defaultView: 'grid',
            grid: {
                layout: [
                    { heading: 'Name', name: 'name', gridTag: 'h2' }
                ],
                showId: true
            }
        },
        itemProps: [
            {
                name: 'name',
                label: 'Text',
                type: 'text',
                required: true,
                minLength: 3,
                maxLength: 30
            }
        ]
    }
};

it('GridView matches snapshot', () => {
    const wrapper = shallow(<GridViewCard {...props} />);
    expect(wrapper).toMatchSnapshot();
});