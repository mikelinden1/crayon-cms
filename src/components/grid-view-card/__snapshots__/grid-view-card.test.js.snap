// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridView matches snapshot 1`] = `
<div
  className="card border-secondary"
  key="grid-item-1"
>
  <div
    className="card-body text"
    key="grid-item-prop-name"
  >
    <h2>
      <Connect(RenderPropValue)
        column={
          Object {
            "gridTag": "h2",
            "heading": "Name",
            "name": "name",
          }
        }
        item={
          Object {
            "id": 1,
            "name": "item 1",
          }
        }
      />
    </h2>
  </div>
  <div
    className="card-footer text-right"
  >
    <div
      className="grid-item-id"
    >
      #
      1
    </div>
    <ActionBtns
      item={
        Object {
          "id": 1,
          "name": "item 1",
        }
      }
    />
  </div>
</div>
`;
